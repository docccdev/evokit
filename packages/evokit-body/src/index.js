import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
