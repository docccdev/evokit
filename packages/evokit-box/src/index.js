// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
