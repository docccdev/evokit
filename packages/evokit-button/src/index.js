import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
