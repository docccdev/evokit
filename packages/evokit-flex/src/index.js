import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'display',
    'content',
    'items',
    'direction',
    'wrap',
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'order',
]);

Flex.Item = FlexItem;
