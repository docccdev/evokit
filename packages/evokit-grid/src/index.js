import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
