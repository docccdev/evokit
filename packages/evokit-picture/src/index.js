import { createBlock } from 'evokit';

export const Picture = createBlock('picture', 'picture', [
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);

export const PictureItem = createBlock('img', 'picture__item', ['display']);

Picture.Item = PictureItem;
