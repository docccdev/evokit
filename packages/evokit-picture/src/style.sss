@import './mixins.sss'

@custom-media --ek-picture-media-small only screen and (min-width: 480px)
@custom-media --ek-picture-media-medium only screen and (min-width: 768px)
@custom-media --ek-picture-media-large only screen and (min-width: 960px)
@custom-media --ek-picture-media-wide only screen and (min-width: 1200px)
@custom-media --ek-picture-media-huge only screen and (min-width: 1400px)

:root
    // prop 'picture-round'
    --ek-picture-round-xxs: 2px
    --ek-picture-round-xs: 4px
    --ek-picture-round-s: 6px
    --ek-picture-round-m: 8px
    --ek-picture-round-l: 10px
    --ek-picture-round-xl: 12px
    --ek-picture-round-xxl: 14px
    --ek-picture-round-3xl: 16px
    --ek-picture-round-4xl: 18px
    --ek-picture-round-5xl: 20px
    // prop 'picture-border-width'
    --ek-picture-border-width-xxs: 1px
    --ek-picture-border-width-xs: 2px
    --ek-picture-border-width-s: 3px
    --ek-picture-border-width-m: 4px
    --ek-picture-border-width-l: 5px
    --ek-picture-border-width-xl: 6px
    --ek-picture-border-width-xxl: 7px
    --ek-picture-border-width-3xl: 8px
    --ek-picture-border-width-4xl: 9px
    --ek-picture-border-width-5xl: 10px

.picture
    display: block
    position: relative
    overflow: hidden
    box-sizing: border-box
    border: transparent 0 solid

    &:after
        content: ""
        display: block
        padding-bottom: 100%

.picture__item
    position: absolute
    top: 0
    right: 0
    bottom: 0
    left: 0
    max-width: 100%
    max-height: 100%
    margin: auto
    outline: none
    border: 0

    &[alt]
        word-wrap: break-word
        word-break: break-word
        hyphens: auto

/**
 * Display:
 *
 *  - picture_display_block
 *  - picture_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .picture_display_$(value)
        display: $value

/**
 * Item display:
 *
 *  - picture__item_display_block
 *  - picture__item_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .picture__item_display_$(value)
        display: $value

/**
 * Fit:
 *
 *  - picture_fit_*
 *
 **/

@each $value in (none, fill, contain, cover, scale-down)
    @mixin media .picture_fit_$(value)
        transform: translatez(0)

        & > .picture__item
            object-fit: $value
            width: 100%
            height: 100%

/**
 * Round:
 *
 *  - picture_round_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl, full)
    @mixin media .picture_round_$(value)
        @if $value == none
            border-radius: 0
        @else if $value == full
            border-radius: 50%
        @else
            border-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-top_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-top-right-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-top-right-radius: 50%
        @else
            border-top-left-radius: var(--ek-picture-round-$(value))
            border-top-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-right_$(value)
        @if $value == none
            border-top-right-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-picture-round-$(value))
            border-bottom-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-bottom_$(value)
        @if $value == none
            border-bottom-left-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-picture-round-$(value))
            border-bottom-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-left_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-bottom-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-bottom-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-picture-round-$(value))
            border-bottom-left-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-top-left_$(value)
        @if $value == none
            border-top-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-top-right_$(value)
        @if $value == none
            border-top-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-bottom-right_$(value)
        @if $value == none
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-right-radius: 50%
        @else
            border-bottom-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-bottom-left_$(value)
        @if $value == none
            border-bottom-left-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-picture-round-$(value))

/**
 * Border width:
 *
 *  - picture_border-width_none
 *  - picture_border-width_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl)
    @mixin media .picture_border-width_$(value)
        @if $value == none
            border-width: 0
        @else
            border-width: var(--ek-picture-border-width-$(value))

    @mixin media .picture_border-width-top_$(value)
        @if $value == none
            border-top-width: 0
        @else
            border-top-width: var(--ek-picture-border-width-$(value))

    @mixin media .picture_border-width-right_$(value)
        @if $value == none
            border-right-width: 0
        @else
            border-right-width: var(--ek-picture-border-width-$(value))

    @mixin media .picture_border-width-bottom_$(value)
        @if $value == none
            border-bottom-width: 0
        @else
            border-bottom-width: var(--ek-picture-border-width-$(value))

    @mixin media .picture_border-width-left_$(value)
        @if $value == none
            border-left-width: 0
        @else
            border-left-width: var(--ek-picture-border-width-$(value))

/**
 * Border style:
 *
 *  - picture_border-style_solid
 *  - picture_border-style_dotted
 *  - picture_border-style_dashed
 *
 **/

@each $value in (solid, dotted, dashed)
    @mixin media .picture_border-style_$(value)
        border-style: $value

    @mixin media .picture_border-style-top_$(value)
        border-top-style: $value

    @mixin media .picture_border-style-right_$(value)
        border-right-style: $value

    @mixin media .picture_border-style-bottom_$(value)
        border-bottom-style: $value

    @mixin media .picture_border-style-left_$(value)
        border-left-style: $value
