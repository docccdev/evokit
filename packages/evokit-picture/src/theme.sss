@import './mixins.sss'

:root
    --ek-picture-border-TEMPLATE_NAME: var(--border)

@mixin media .picture_border_TEMPLATE_NAME
    border: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-top_TEMPLATE_NAME
    border-top: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-right_TEMPLATE_NAME
    border-right: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-bottom_TEMPLATE_NAME
    border-bottom: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-left_TEMPLATE_NAME
    border-left: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-tb_TEMPLATE_NAME
    border-top: 1px solid var(--ek-picture-border-TEMPLATE_NAME)
    border-bottom: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-lr_TEMPLATE_NAME
    border-left: 1px solid var(--ek-picture-border-TEMPLATE_NAME)
    border-right: 1px solid var(--ek-picture-border-TEMPLATE_NAME)
