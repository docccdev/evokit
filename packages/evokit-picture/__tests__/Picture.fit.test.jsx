import React from 'react';
import { shallow } from 'enzyme';
import { Picture } from '../src';

describe('<Picture picture-fit="..." />', () => {
    it('props picture-fit', () => {
        const wrapper = shallow(<Picture picture-fit='value' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<picture class="ek-picture ek-picture_fit_value"></picture>');
    });
});
