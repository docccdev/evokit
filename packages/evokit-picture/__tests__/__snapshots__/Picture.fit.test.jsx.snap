// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picture picture-fit="..." /> props picture-fit 1`] = `
<picture
  className="ek-picture ek-picture_fit_value"
/>
`;
